src/Associations.cc

```diff
@@ -139,34 +140,64 @@ void Associations::associateCatalogs(const double matchCutInArcSec, const bool u
 }
 
 void Associations::collectRefStars(lsst::afw::table::SortedCatalogT<lsst::afw::table::SimpleRecord> &refCat,
-                                   afw::geom::Angle matchCut, std::string const &fluxField) {
+                                   afw::geom::Angle matchCut, std::string const &fluxField,
+                                   std::map<std::string, std::vector<double> > const &refFluxMap,
+                                   std::map<std::string, std::vector<double> > const &refFluxErrMap) {
     if (refCat.size() == 0) {
         throw(LSST_EXCEPT(pex::exceptions::InvalidParameterError,
                           " reference catalog is empty : stop here "));
     }
 
-    //  auto coordKey = refCat.getSchema().find<lsst::afw::coord::Coord>("coord").key;
-    // Same syntax as the following line but with auto :  auto coordKey =
-    // afwTable::CoordKey(refCat.getSchema()["coord"]);
     afw::table::CoordKey coordKey = refCat.getSchema()["coord"];
     auto fluxKey = refCat.getSchema().find<double>(fluxField).key;
+    // Don't blow up if the reference catalog doesn't contain errors.
+    afw::table::Key<double> fluxErrKey;
+    try {
+        fluxErrKey = refCat.getSchema().find<double>(fluxField + "Sigma").key;
+    } catch (pex::exceptions::NotFoundError) {
+        LOGLS_WARN(_log, "Flux error field ("
+                                 << fluxField + "Sigma"
+                                 << ") not found in reference catalog. Not using ref flux errors.");
```

Comment:
Indentation is what clang-format gave. The `+` is because I copied that from just above it. Should I make that `<<`?

I was actually expecting you to comment on my use of `try:catch` here, as I'm not sure this usage is generally accepted in C++.